addons/resource/models/resource_calendar.py

```diff
@@ -368,10 +368,15 @@ def _attendance_intervals_batch(self, start_dt, end_dt, resources=None, domain=N
             res_intervals = WorkIntervals(res)
             for resource in resources:
                 if resource and resource._is_flexible():
-                # If the resource is flexible, return the whole period from start_dt to end_dt with a dummy attendance
+                    duration_days = (end - start).days + (0.5 if (end - start).total_seconds() / 3600 < resource.calendar_id.hours_per_day else 1)
+                    if resource.calendar_id and resource.calendar_id.flexible_hours:
+                        duration_hours = duration_days * resource.calendar_id.hours_per_day
+                    elif resource._is_fully_flexible():
```

Comment:
A resource without calendar always is fully flexible + leaving the conditions without a "else" to declare a variable used below might result in a crash if it ends up undeclared (though in this specific case it shouldn't) 
```suggestion
                    else:
```